android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt

```diff
@@ -4,25 +4,67 @@ import androidx.fragment.app.FragmentActivity
 import com.revenuecat.purchases.Offering
 
 @JvmOverloads
+@Deprecated(
+    message = "Use presentPaywallFromFragment with Options instead",
+    replaceWith = ReplaceWith(
+        expression = "presentPaywallFromFragment(fragment, options)",
+        imports = ["com.revenuecat.purchases.hybridcommon.ui.presentPaywallFromFragment"],
+    ),
+)
 fun presentPaywallFromFragment(
     fragment: FragmentActivity,
     requiredEntitlementIdentifier: String? = null,
     paywallResultListener: PaywallResultListener? = null,
     shouldDisplayDismissButton: Boolean? = null,
     offering: Offering? = null,
 ) {
-    fragment
-        .supportFragmentManager
-        .beginTransaction()
-        .add(
-            PaywallFragment.newInstance(
-                fragment,
-                requiredEntitlementIdentifier,
-                paywallResultListener,
-                shouldDisplayDismissButton,
-                offering,
-            ),
-            PaywallFragment.tag,
-        )
-        .commit()
+    presentPaywallFromFragment(
+        fragment,
+        PresentPaywallOptions(
+            requiredEntitlementIdentifier = requiredEntitlementIdentifier,
+            paywallResultListener = paywallResultListener,
+            shouldDisplayDismissButton = shouldDisplayDismissButton,
+            offering = offering,
+        ),
+    )
 }
+
+fun presentPaywallFromFragment(
+    fragment: FragmentActivity,
+    options: PresentPaywallOptions,
+) {
+    with(options) {
+        fragment
+            .supportFragmentManager
+            .beginTransaction()
+            .add(
+                if (offeringIdentifier != null) {
+                    PaywallFragment.newInstance(
+                        fragment,
+                        requiredEntitlementIdentifier,
+                        paywallResultListener,
+                        shouldDisplayDismissButton,
+                        offeringIdentifier,
+                    )
+                } else {
+                    PaywallFragment.newInstance(
+                        fragment,
+                        requiredEntitlementIdentifier,
+                        paywallResultListener,
+                        shouldDisplayDismissButton,
+                        offering,
+                    )
+                },
+                PaywallFragment.tag,
+            )
+            .commit()
+    }
+}
+
+data class PresentPaywallOptions(
+    val requiredEntitlementIdentifier: String? = null,
+    val paywallResultListener: PaywallResultListener? = null,
+    val shouldDisplayDismissButton: Boolean? = null,
+    val offering: Offering? = null,
```

Comment:
Hmm could we just entirely remove this one in favor of `offeringIdentifier`? If it's for the deprecated method, we can just pass in the id there I think.

Reply:
Wouldn't that mean we have to fetch offerings again potentially? 

Can we define a type like this:
```kotlin
sealed class PaywallSource {
    class Offering(val value: PurchasesOffering): PaywallSource()
    class DefaultOffering()
    class OfferingIdentifier(val value: String): PaywallSource()
}
```

Like I did in iOS, to make the 3 options type safe?

Reply:
Right... Yeah makes sense. We have something similar as an internal API in native android: https://github.com/RevenueCat/purchases-android/blob/main/ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallOptions.kt#L6... But I don't know if we want to expose that. But yeah, in here that could make sense.

Reply:
> Wouldn't that mean we have to fetch offerings again potentially?

The hybrids are always going to use the offering identifier since there's no way to send a whole offering from JS, Dart, etc to the native side

Reply:
But I think I am going to add the type just in case

Reply:
Yes that's what I'm doing in the opened PR, but this helper in phc will always get an offering identifier, it will never be an offering

Reply:
Yup yup 👍🏻